src/main/java/controllers/FeedController.java

```diff
@@ -30,56 +29,171 @@ public FeedController(InOut inOut, LoginManager loginManager) {
     }
 
     public void run(ShellAction action) {
-        // TODO: get user inputs/display info based on the ShellAction
         if (action == ShellAction.BROWSEFEED) {
             this.runBrowseFeed();
         }
     }
 
     private void runBrowseFeed() {
+        // Ask for a filter option for the feed
+        ArrayList<Integer> filterInput = this.getFilterInput();
+        this.feedManager.setFeedFilter(filterInput);
         Feed feed = this.feedManager.getCurrentUsersFeed();
-        ArrayList<Post> allPosts = feed.getPosts();
-        this.postsString = this.generateFeedActions(allPosts);
-        try {
-            String postSelection = this.inOut.getInput(this.postsString);
-            Integer postNumber = Integer.parseInt(postSelection);
-            Post selectedPost = this.postsActionMap.get(postNumber);
-            this.displayPostInfo(selectedPost);
-            String postAction =  this.inOut.getInput(this.postActionPrompt);
-            this.runPostAction(selectedPost, Integer.parseInt(postAction));
-            this.displayPostInfo(selectedPost);
-        } catch (IOException e) {
-            this.inOut.setOutput("An error occurred: " + e);
-        } catch (NumberFormatException nfe) {
-            this.inOut.setOutput("You entered an invalid action input.");
-        }
 
+        // Generate the posts and ask to select one to like/comment
+        String postsString = this.generateFeedActions(feed.getDisplayedPosts());
+        this.selectOnePost(postsString, feed.getDisplayedPosts().size());
     }
 
     private void runPostAction(Post selectedPost, Integer postAction) {
         switch (postAction) {
             case 0:
                 selectedPost.addLike(this.loginManager.getCurrUser());
                 break;
+            case 1:
```

Comment:
I'm not sure if this switch statement generates an Intellij warning, but if it does, please resolve the warning